### ISSUE TITLE
[mingw-w64-x86_64-tclx] Error loading tclx86.dll

### ISSUE THREAD

**Issue by mayd** [msys2/MINGW-packages#8066]
When I try to load Tcl package TclX in MSYS2/MINGW64 tclsh I see the following console output:

```
$ tclsh
package require Tclx
couldn't load library "C:/msys64/mingw64/lib/tclx8.6/tclx86.dll": this library or a dependent library could not be found in library path
```

Looking at the MSYS2 package inventory for mingw-w64-x86_64-tclx I see the following item on the list:

`/mingw64/bin/tclx86.dll`

This suggested to me that the required file is installed in the wrong directory.

After copying tclx86.dll to /mingw64/lib/tclx8.6/tclx86.dll  and rerunning the commands the error does not occur:

```
$  tclsh
package require Tclx

```

**Comment by mayd** [comment]
PS the same error occurs for package mingw-w64-i686-tclx and was fixed in the same way.

**Comment by Biswa96** [comment]
Not sure about your issue but both tclsh works in my setup. Here are the dependencies:

```
$ ntldd /mingw64/bin/tclsh.exe
        KERNEL32.dll => C:\Windows\SYSTEM32\KERNEL32.dll (0x0000000000640000)
        msvcrt.dll => C:\Windows\SYSTEM32\msvcrt.dll (0x0000000001430000)
        tcl86.dll => C:\msys64\mingw64\bin\tcl86.dll (0x0000000000d60000)

$ ntldd /usr/bin/tclsh.exe
        libtcl8.6.dll => C:\msys64\usr\bin\libtcl8.6.dll (0x0000000000bb0000)
        msys-2.0.dll => C:\msys64\usr\bin\msys-2.0.dll (0x0000000000e40000)
        KERNEL32.dll => C:\Windows\SYSTEM32\KERNEL32.dll (0x00000000011b0000)
```

**Comment by Amphaal** [comment]
`pkgIndex.tcl` of `mingw-w64-x86_64-tclx` contains : 

```
package ifneeded Tclx 8.6 \
	[list load [file join $dir tclx86.dll] Tclx]
```

I think `$dir` there should be changed to another variable pointing to the mingw64 bindir.

**Comment by mayd** [comment]
Could someone at least RUN my minimal test case before dismissing the bug?

This bug does is not triggered by simply running tclsh - it occurs when you attempt to **load package Tclx**.

I am unclear about current Tcl behaviour but according to the documented standards for installing Tcl extensions
[https://github.com/tcltk/sampleextension](https://github.com/tcltk/sampleextension) 

"The main .so|.dll library file gets installed in the versioned PACKAGE
directory, which is OK on all platforms because it will be directly
referenced with by 'load' in the pkgIndex.tcl file.  Dependent DLL files on
Windows **must go in the bin directory (or other directory on the user's
PATH) in order for them to be found**."

**Comment by Biswa96** [comment]
> it occurs when you attempt to load package Tclx

You did not add the prompt character (%) in your actual question. It means a lot. Yes, your issue can be reproduced now. Maybe some anomaly in autoconf. Wondering why this has not be reported so long.

**Comment by mayd** [comment]
There is no % prompt when tclsh is launched from MSYS2 bash.  That is another bug I was going to report.  
There seems to be major problem with writing to the console in tclsh and wish shells.

**Comment by Biswa96** [comment]
> There is no % prompt when tclsh is launched from MSYS2 bash.

Well, we have another candidate for enable ConPTY. `MSYS=enable_pcon` shows the prompt text in mingw tclsh one.

**Comment by mati865** [comment]
Alternatively run these commands through `winpty` for now.

**Comment by mayd** [comment]
Alternatives to bash fix the console I/O issue but that is unrelated to the reported bug regarding TclX error.

**Comment by mayd** [comment]
I started to rebuild the package from source and identified several issues. The most obvious being the 
following line in the PKGBUILD script:

`  mv ${pkgdir}${MINGW_PREFIX}/lib/tclx8.6/*.dll ${pkgdir}${MINGW_PREFIX}/bin/`

**Comment by mayd** [comment]
I have fixed this bug plus corrected missing --tcl-prefix option for configure script for this package.
See Github pull request #8090.

**Comment by mayd** [comment]
I upgraded my MSYS2 installation from aMSYS2 internet mirror via pacman. 

Confirmed packages mingw-w64-i686-tclx and mingw-w64-x86_64-tclx were upgraded to versions mingw-w64-i686-tclx 8.4.4-2 and mingw-w64-x86_64-tclx 8.4.4-2.

Confirmed that the Tclx package loads correctly in both MINGW32 and MINGW64 environments.

I fetched the MINGW Github repository source for MINGW-packages.

Confirmed package versions mingw-w64-i686-tclx 8.4.4-2 and mingw-w64-x86_64-tclx 8.4.4-2 downloaded and built without errors.  I also ran the Tclx test suite for both packages and obtained expected test results.

I believe this issue is fixed.
